cpu-sim/src/lib.rs

```diff
@@ -223,9 +224,9 @@ mod tests {
             fifo.extend_from_slice(&bytes);
         };
 
-        // Run the simulation with FIFO callback
-        // The program writes "Hello World!" to the FIFO TX
-        let result = run_elf_with_callback(&elf_path, 10000, false, Some(callback))
+        // Run the simulation with FIFO callback and RX data
+        // The program should echo "TestData" from RX to TX
+        let result = run_elf_with_fifo(&elf_path, 10000, false, Some(callback), Some("TestData"))
```

Comment:
Please attempt to debug the echo issue further by enabling the instruction trace printing feature here and verifying that the test executes the expected sequence of assembly instructions with the correct data values.